packages/teams-js/src/internal/utils.ts

```diff
@@ -266,3 +267,25 @@ export function callCallbackWithErrorOrResultOrNullFromPromiseAndReturnPromise<T
   });
   return p;
 }
+
+export function createTeamsAppLink(params: pages.NavigateToAppParams): string {
+  let url =
```

Comment:
Wouldn't all of this be much easier using native `URL` type? It's supported in all the places we support, it'll allow you to skip the encoding part and tracking if anything was added or not so you can do the right `?` vs `&` thing.

Reply:
When I looked into this a few weeks ago it appeared that the native `URL` type didn't support this, but perhaps I found old information. If there's a built-in way to build a URL I'd much prefer to use that.